streams/src/main/java/org/apache/kafka/streams/state/internals/metrics/RocksDBMetrics.java

```diff
@@ -0,0 +1,315 @@
+                                                         final String storeName) {
+        final Sensor sensor = streamsMetrics
+            .storeLevelSensor(taskName, storeName, BYTES_READ_DURING_COMPACTION, RecordingLevel.DEBUG);
+        addAmountRateMetricToSensor(sensor,
+                                    STATE_LEVEL_GROUP,
+                                    streamsMetrics.storeLevelTagMap(taskName, STORE_TYPE_PREFIX + storeType, storeName),
+                                    BYTES_READ_DURING_COMPACTION,
+                                    "Average per-second number of bytes read during compaction");
+        return sensor;
+    }
+
+    public static Sensor bytesWrittenDuringCompactionSensor(final StreamsMetricsImpl streamsMetrics,
+                                                            final String taskName,
+                                                            final String storeType,
+                                                            final String storeName) {
+        final Sensor sensor = streamsMetrics
+            .storeLevelSensor(taskName, storeName, BYTES_WRITTEN_DURING_COMPACTION, RecordingLevel.DEBUG);
+        addAmountRateMetricToSensor(sensor,
+                                    STATE_LEVEL_GROUP,
+                                    streamsMetrics.storeLevelTagMap(taskName, STORE_TYPE_PREFIX + storeType, storeName),
+                                    BYTES_WRITTEN_DURING_COMPACTION,
+                                    "Average per-second number of bytes written during compaction");
+        return sensor;
+    }
+
+    public static Sensor compactionTimeAvgSensor(final StreamsMetricsImpl streamsMetrics,
+                                                 final String taskName,
+                                                 final String storeType,
+                                                 final String storeName) {
+        final Sensor sensor = streamsMetrics
+            .storeLevelSensor(taskName, storeName, COMPACTION_TIME + AVG_SUFFIX, RecordingLevel.DEBUG);
+        addValueMetricToSensor(sensor,
+                               STATE_LEVEL_GROUP,
+                               streamsMetrics.storeLevelTagMap(taskName, STORE_TYPE_PREFIX + storeType, storeName),
+                               COMPACTION_TIME + AVG_SUFFIX,
+                               "Average time spent on compaction in ms");
+        return sensor;
+    }
+
+    public static Sensor compactionTimeMinSensor(final StreamsMetricsImpl streamsMetrics,
+                                                 final String taskName,
+                                                 final String storeType,
+                                                 final String storeName) {
+        final Sensor sensor = streamsMetrics
+            .storeLevelSensor(taskName, storeName, COMPACTION_TIME + MIN_SUFFIX, RecordingLevel.DEBUG);
+        addValueMetricToSensor(sensor,
```

Comment:
I'm a bit confused here for max / min we are creating a sensor with type `Value` still, but since I did not see how the `record()` is going to be invoked maybe I'm just wrong.. otherwise, could we just use a similar function like `addAvgAndMax`?

Reply:
Just chiming in, I think we're just defining the metrics in this PR and then actually recording them later. We might change the actual metric type later on when we look at how it's going to actually get recorded, so maybe we shouldn't worry too much about the selected metric implementation right now.

Reply:
Agreed